From a657e7dca02c7e94dc3f1e8b5dc8e92e57894700 Mon Sep 17 00:00:00 2001
From: Alexander Mikhalitsyn <alexander.mikhalitsyn@virtuozzo.com>
Date: Fri, 5 Aug 2022 08:32:00 +0200
Subject: [PATCH 3/5] UBUNTU: SAUCE: overlayfs: remove CONFIG_AUFS_FS
 dependency

BugLink: https://bugs.launchpad.net/bugs/1983640

Right now we have a fix:
b07bc17b ("UBUNTU: SAUCE: overlayfs: fix incorrect mnt_id of files opened from map_files")
in master branch of Jammy kernel, but only formaly. Because these kernels compiled without
CONFIG_AUFS_FS set, so this fix just disabled. There is no need to make it dependent on
CONFIG_AUFS_FS option, because in all cases we have mm/prfile.c compiled-in.

Fixes: b07bc17b ("UBUNTU: SAUCE: overlayfs: fix incorrect mnt_id of files opened from map_files")
Signed-off-by: Alexander Mikhalitsyn <alexander.mikhalitsyn@virtuozzo.com>
Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
Acked-by: Stefan Bader <stefan.bader@canonical.com>
Acked-by: Kleber Sacilotto de Souza <kleber.souza@canonical.com>
Signed-off-by: Stefan Bader <stefan.bader@canonical.com>
---
 fs/overlayfs/file.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/fs/overlayfs/file.c b/fs/overlayfs/file.c
index a60cfb20630e..33e3ec609fa9 100644
--- a/fs/overlayfs/file.c
+++ b/fs/overlayfs/file.c
@@ -488,7 +488,6 @@ static int ovl_fsync(struct file *file, loff_t start, loff_t end, int datasync)
 	return ret;
 }
 
-#if IS_ENABLED(CONFIG_AUFS_FS)
 /*
  * In map_files_get_link() (fs/proc/base.c)
  * we need to determine correct path from overlayfs.
@@ -514,12 +513,6 @@ static void ovl_vm_prfile_set(struct vm_area_struct *vma,
 	vma->vm_region->vm_prfile = file;
 #endif
 }
-#else /* !CONFIG_AUFS_FS */
-static void ovl_vm_prfile_set(struct vm_area_struct *vma,
-			      struct file *file)
-{
-}
-#endif/* CONFIG_AUFS_FS */
 
 static int ovl_mmap(struct file *file, struct vm_area_struct *vma)
 {
-- 
2.34.1

